tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/XUnit/XUnitTestRunnerRunAsyncIntegration.cs

```diff
@@ -36,8 +37,15 @@ internal static CallTargetState OnMethodBegin<TTarget>(TTarget instance)
             {
                 TestRunnerStruct runnerInstance = instance.DuckCast<TestRunnerStruct>();
 
+                // Check if the test should be skipped by the ITR
+                if (XUnitIntegration.ShouldSkip(ref runnerInstance) && instance.TryDuckCast<ITestRunnerSkippeable>(out var skippeableRunnerInstance))
+                {
+                    Common.Log.Debug("ITR: Test skipped: {class}.{name}", runnerInstance.TestClass.FullName, runnerInstance.TestMethod.Name);
+                    skippeableRunnerInstance.SkipReason = "Skipped by the Intelligent Test Runner";
+                }
+
                 // Skip test support
-                if (runnerInstance.SkipReason != null)
+                if (runnerInstance.SkipReason != null && instance is not null)
```

Comment:
`instance` can't be null if we got this far, it would have blown up earlier.

Reply:
I moved the check on https://github.com/DataDog/dd-trace-dotnet/pull/3075/commits/bf59726334fcce41d24ef791a50d1bf8dea47a72